pkg/agent/agent.go

```diff
@@ -174,6 +174,11 @@ func (a *Agent) initEndpoints() error {
 		return fmt.Errorf("Error creating GRPC listener: %s", err)
 	}
 
+	if addr.Network() == "unix" {
+		// Any process should be able to use this unix socket
+		os.Chmod(addr.String(), os.ModePerm)
+	}
```

Comment:
non-blocking: Now that we have a handful of statements which deal with creating a listener, it may make sense to introduce a `createListener` method or something similar

Reply:
Do you think that that should be addressed as part of this issue?

Reply:
Nope doesn't need to be :) feel free to merge regardless